packages/flutter/lib/src/material/material_localizations.dart

```diff
@@ -112,14 +112,23 @@ abstract class MaterialLocalizations {
 
   /// Formats [TimeOfDay.hour] in the given time of day according to the value
   /// of [timeOfDayFormat].
-  String formatHour(TimeOfDay timeOfDay);
+  ///
+  /// If [alwaysUse24HourFormat] is true, formats hour using [HourFormat.HH]
+  /// rather than the default for the current locale.
+  String formatHour(TimeOfDay timeOfDay, { bool alwaysUse24HourFormat: false });
 
   /// Formats [TimeOfDay.minute] in the given time of day according to the value
   /// of [timeOfDayFormat].
   String formatMinute(TimeOfDay timeOfDay);
 
   /// Formats [timeOfDay] according to the value of [timeOfDayFormat].
-  String formatTimeOfDay(TimeOfDay timeOfDay);
+  ///
+  /// If [alwaysUse24HourFormat] is true, formats hour using [HourFormat.HH]
+  /// rather than the default for the current locale. This value is usually
+  /// passed from [MediaQueryData.alwaysUse24HourFormat], and it has platform-
+  /// specific behavior. For more details, see the documentation for
+  /// [MediaQueryData.alwaysUse24HourFormat].
```

Comment:
if you feel it is important to link to MediaQueryData.alwaysUse24HourFormat twice, consider the more common "see also" style, as in:

```
  /// See also:
  ///
  ///  * [MediaQueryData.alwaysUse24HourFormat], the documentation for which provides more detail.
```
...or something.

Reply:
(or leave it as you have it now if you think that's clearer)